snippets/header-dropdown-menu.liquid

```diff
@@ -7,7 +7,22 @@
 
 <nav class="header__inline-menu">
   <ul class="list-menu list-menu--inline" role="list">
-    {%- for link in section.settings.menu.links -%}
+         {% assign showB2BNav = false %}
+      {%- if showB2BNav == true -%}
+
+        {%- if customer.metafields.sparklayer.authentication -%}
+          {% assign menu = linklists.b2b-menu %}
+        {%- else -%}
+          {% assign menu = linklists[section.settings.menu] %}
+          
+        {%- endif -%}
+
+    {%- else -%}
+
+    {% assign menu = linklists[section.settings.menu] %}
+    {%- endif -%}
+
+    {%- for link in menu.links -%}
```

Comment:
```suggestion
    {%- if customer.metafields.sparklayer.authentication -%}
      {% assign menu = linklists.b2b-menu | default: linklists[section.settings.menu] %}
    {%- else -%}
      {% assign menu = linklists[section.settings.menu] %}
    {%- endif -%}
    {%- for link in menu -%}
```
Simples, if the b2b menu doesn't exist, fall back to default menu, needs testing though, see [here](https://shopify.github.io/liquid/filters/default/) how to use